src/Item/Item.router.ts

```diff
@@ -207,12 +211,18 @@ export class ItemRouter extends Router {
     const allItems = allItemsWithCount.map((itemWithCount) =>
       omit<ItemAttributes>(itemWithCount, ['total_count'])
     )
+    const collectionIds = allItems.map((item) => item.collection_id)
     const { items: dbItems, tpItems: dbTPItems } = this.itemService.splitItems(
       allItems
     )
 
     const [items, tpItems] = await Promise.all([
-      Bridge.consolidateItems(dbItems, remoteItems),
+      Bridge.consolidateItems(
+        dbItems,
+        remoteItems.filter((remoteItem) =>
+          collectionIds.includes(remoteItem.collection.id)
+        )
```

Comment:
There shouldn't be any remote items that don't have collections, is there a reason for this filter?

Reply:
Since we're adding a filter by `collectionId`, at the moment of consolidating the items, we should only use the `remoteItems` assign to that collectionId because later on in `consolidateItems` both `dbItems` and the dbItems from the remoteItems are merged. 
Imagine this case:
```
dbItems = [{ id: 1, collectionId: null} , { id: 2, collectionId: null} ]
remoteItems = [{ id: 3, collectionId: 'id'} ,{ id: 4, collectionId: 'id'}] 
```
if we don't filter the remoteItems, the output of the `consolidateItems` will include the ids `3` and `4`.